src/wallet.rs

```diff
@@ -194,6 +205,175 @@ impl Wallet {
+    shard_config: Option<ShardConfig>,
+}
+
+impl Builder {
+    pub fn new() -> Builder {
+        Builder {
+            output: PathBuf::from("wallet.key"),
+            password: Default::default(),
+            force: false,
+            seed_type: None,
+            seed_words: None,
+            key_tag: Default::default(),
+            shard_config: None,
+        }
+    }
+
+    /// Sets the output file for the wallet.
+    /// Defaults to 'main.key'
+    pub fn output(mut self, path: &Path) -> Builder {
+        self.output = path.to_path_buf();
+        self
+    }
+
+    /// Sets the wallet's password
+    /// Defaults to '' (empty string)
+    pub fn password(mut self, pwd: &str) -> Builder {
+        self.password = pwd.to_owned();
+        self
+    }
+
+    /// Force overwrite of wallet if output already exists
+    /// Defaults to false
+    pub fn force(mut self, overwrite: bool) -> Builder {
+        self.force = overwrite;
+        self
+    }
+
+    /// Set the seed type if using seed words to create this wallet
+    /// Defaults to None
+    pub fn seed_type(mut self, seed_type: Option<SeedType>) -> Builder {
+        self.seed_type = seed_type;
+        self
+    }
+
+    /// The seed words used to create this wallet
+    /// Defaults to None
+    pub fn seed_words(mut self, seed_words: Option<Vec<String>>) -> Builder {
+        self.seed_words = seed_words;
+        self
+    }
+
+    /// The type of key to generate (ecc_compact/ed25519)
+    /// Defaults to ed25519
+    pub fn key_tag(mut self, key_tag: &KeyTag) -> Builder {
+        self.key_tag = *key_tag;
+        self
+    }
+
+    /// Optional shard config info to use in order to create a sharded wallet
+    /// otherwise, creates a basic non-sharded wallet
+    pub fn shard_config(mut self, shard_config: Option<ShardConfig>) -> Builder {
+        self.shard_config = shard_config;
+        self
+    }
+
+    /// Creates a new wallet
+    pub fn create(self) -> Result<Wallet> {
+        let keypair = gen_keypair(self.key_tag, self.seed_words, self.seed_type.as_ref())?;
+
+        let wallet = if let Some(shard_config) = self.shard_config {
+            let format = format::Sharded {
+                key_share_count: shard_config.key_share_count,
+                recovery_threshold: shard_config.recovery_threshold,
+                pwhash: PwHash::argon2id13_default(),
+                key_shares: vec![],
+            };
+            Wallet::encrypt(&keypair, self.password.as_bytes(), Format::Sharded(format))?
+        } else {
+            let format = format::Basic {
+                pwhash: PwHash::argon2id13_default(),
+            };
+            Wallet::encrypt(&keypair, self.password.as_bytes(), Format::Basic(format))?
+        };
+
+        if self.shard_config.is_some() {
+            let extension = self
+                .output
+                .extension()
+                .unwrap_or_else(|| OsStr::new(""))
```

Comment:
this seems forced? What is this trying to achieve?

Reply:
This was copy of what was already there. Didn't want to refactor what was already working. (https://github.com/helium/helium-wallet-rs/blob/9f51a08eb5b95c24fd06391dab733df96df43a82/src/cmd/mod.rs#L201)